pkg/handlers/v1/attribute_test.go

```diff
@@ -0,0 +1,106 @@
+package v1
+
+import (
+	"context"
+	"errors"
+	"io/ioutil"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/asecurityteam/logevent"
+	"github.com/asecurityteam/nexpose-asset-attributor/pkg/assetattributor"
+	"github.com/asecurityteam/nexpose-asset-attributor/pkg/domain"
+)
+
+func TestSuccess(t *testing.T) {
+	input := domain.NexposeAssetVulnerabilities{
+		Asset: domain.Asset{
+			ID: 123,
+		},
+	}
+
+	handler := &AttributeHandler{
+		AssetAttributor: assetattributor.NewNoOpAssetAttributor(),
+		LogFn:           domain.LoggerFromContext,
```

Comment:
I made this same comment on another PR somewhere but I can't quite remember where...

Using the actual log and stat client functions in tests is inconsistent with how we generally handle code dependencies with mocks. The current implementation of the stat client is `xstats` which just happens to return a `NOP` version if the context is empty. The current implementation of the log client does not do this which is why you have to inject one into the context below. The better way to handle this is by passing in a function that satisfies the `LogFn` and `StatFn` that returns a mock or `NOP` version of the interface to use. This will prevent us from needing to refactor these tests if the underlying implementations of those tools change.

Reply:
I was planning on doing it on as-needed basis, but I can go ahead and add that. 👍

Reply:
Done!